[primTorch] Redefine prim.collapse{,_view} end point to be inclusive

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #92017
* #91748

This makes `prims.collapse(a, start, end)` match the behavior of
`torch.flatten(a, start, end)` more closely.

peterbell10 committed:
[primTorch] Redefine prim.collapse{,_view} end point to be inclusive

This makes `prims.collapse(a, start, end)` match the behavior of
`torch.flatten(a, start, end)` more closely.

[ghstack-poisoned]

peterbell10 committed:
lint on "[primTorch] Redefine prim.collapse{,_view} end point to be inclusive"


This makes `prims.collapse(a, start, end)` match the behavior of
`torch.flatten(a, start, end)` more closely.

[ghstack-poisoned]

peterbell10 committed:
Update on "[primTorch] Redefine prim.collapse{,_view} end point to be inclusive"


This makes `prims.collapse(a, start, end)` match the behavior of
`torch.flatten(a, start, end)` more closely.

[ghstack-poisoned]

mruberry reviewed (approved):
Cool! Test failures look unrelated

peterbell10 committed:
Update on "[primTorch] Redefine prim.collapse{,_view} end point to be inclusive"


This makes `prims.collapse(a, start, end)` match the behavior of
`torch.flatten(a, start, end)` more closely.

[ghstack-poisoned]

peterbell10 commented:
@pytorchbot merge
